samcli/commands/check/calculations.py

```diff
@@ -0,0 +1,115 @@
+                "For the lambda function [%s], the %ims duration and %iTPS arrival rate is using %i%% of the allowed "
+                + "concurrency on AWS Lambda. It is very close to the limits of the lambda function. It is strongly "
+                + "recommended that you get a limit increase before deploying your application:"
+                + "\nhttps://console.aws.amazon.com/servicequotas"
+            ) % (resource_name, duration, tps, round(capacity_used))
+            self._graph.red_warnings.append(warning)
+
+        else:  # capacity_used > 100
+            burst_capacity_used = _check_limit(tps, duration, burst_concurrency)
+            warning.message = (
+                "For the lambda function [%s], the %ims duration and %iTPS arrival rate is using %i%% of the allowed "
+                + "concurrency on AWS Lambda. It exceeds the limits of the lambda function. It will use %i%% of the "
+                + "available burst concurrency. It is strongly recommended that you get a limit increase before "
+                + "deploying your application:\nhttps://console.aws.amazon.com/servicequotas"
+            ) % (resource_name, duration, tps, round(capacity_used), round(burst_capacity_used))
+            self._graph.red_burst_warnings.append(warning)
+
+    def run_bottle_neck_calculations(self):
+        """
+        Bottle neck calculations are calculated and stored in the graph
+        """
+        click.echo("Running calculations...")
+
+        for resource in self._graph.resources_to_analyze:
+            resource_type = resource.resource_type
+            resource_name = resource.resource_name
+
+            if resource_type == AWS_LAMBDA_FUNCTION:
+
+                client = boto3.client("service-quotas")
+                burst_concurrency = client.get_aws_default_service_quota(ServiceCode="lambda", QuotaCode="L-548AE339")[
+                    "Quota"
+                ]["Value"]
+                concurrent_executions = client.get_aws_default_service_quota(
+                    ServiceCode="lambda", QuotaCode="L-B99A9384"
+                )["Quota"]["Value"]
+
+                tps = resource.tps
+                duration = resource.duration
+                capacity_used = _check_limit(tps, duration, concurrent_executions)
+
+                self._generate_warning_message(
+                    capacity_used, resource_name, concurrent_executions, duration, tps, burst_concurrency
+                )
+
+
+def _check_limit(tps: int, duration: int, execution_limit: int) -> float:
+    """The check to see if a given resource will cause a bottle neck.
+    Args:
+        tps (int): TPS of lambda function based on parent resource
+        duration (int): Duration of lambda function
+        execution_limit (int): Execution limit based on user account
+
+    Returns:
+        float: percentage of available concurrent executions used
+    """
+    tps_max_limit = (1000 / duration) * execution_limit
```

Comment:
any precautionary measure to make sure duration is not zero?

Reply:
When the user is asked to input duration, they must enter a value within the range 1-900,000 inclusive. Therefore duration can never be zero.